app/controllers/comments_controller.rb

```diff
@@ -15,6 +17,16 @@ def create
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @post = @comment.post
+    @user = @post.author
+    @comment.destroy
+    @post.comments_counter -= 1
+
+    redirect_to user_post_path(@user, @post) if @post.save
```

Comment:
- [ ] Lets add some slight improvement of this part of the code with the one I will add for you below:

```
redirect_to user_post_path(@comment.post.author, @comment.post),
                notice: 'Comment was successfully deleted.'
```

**Doing the steps listed in both inline comments ensures that your delete function works perfectly fine as shown below:**

[Screencast from 20-01-2024 09:00:13 ALASIRI.webm](https://github.com/Sami-Ullah-Tufail/blog/assets/86318284/190799a2-552c-4449-942a-f88b63929c45)